ui/src/components/DashboardMetrics.js

```diff
@@ -0,0 +1,460 @@
+  const [graphWidth, graphHeight] = useDynamicChartSize(
+    'dashboard-metrics-container',
+    1,
+    4,
+  );
+
+  // We use 4 main color from the palette and decline them (lighter/ darker) when we have more than 4 datasets
+  const colorRange = [
+    lineColor1,
+    lineColor2,
+    lineColor3,
+    lineColor4,
+    lighten(0.3, lineColor1),
+    lighten(0.3, lineColor2),
+    lighten(0.3, lineColor3),
+    lighten(0.3, lineColor4),
+    darken(0.2, lineColor1),
+    darken(0.2, lineColor2),
+    darken(0.2, lineColor3),
+    darken(0.2, lineColor4),
+  ];
+
+  const getThroughputColorRange = (size) => {
+    const res = [];
+
+    for (let i = 0; i < size; i++) {
+      if (colorRange[i]) {
+        // Adding twice the same color to have the same line color for both read and write
+        res.push(colorRange[i]);
+        res.push(colorRange[i]);
+      }
+    }
+    return res;
+  };
+
+  const xAxis = {
+    field: 'date',
+    type: 'temporal',
+    axis: {
+      // Refer to all the available time format: https://github.com/d3/d3-time-format#locale_format
+      format:
+        metricsTimeSpan === LAST_TWENTY_FOUR_HOURS ||
+        metricsTimeSpan === LAST_ONE_HOUR
+          ? '%m/%d %H:%M'
+          : '%m/%d',
+      ticks: true,
+      tickCount: 4,
+      labelColor: theme.brand.textSecondary,
+      labelFlush: 20,
+    },
+    title: null,
+  };
+
+  const loadColor = {
+    field: 'type',
+    type: 'nominal',
+    scale: {
+      range: colorRange,
+    },
+    legend: nodes.length
+      ? {
+          direction: 'horizontal',
+          orient: 'bottom',
+          title: null,
+          labelFontSize: 12,
+          symbolSize: 300,
+          columns: 3,
+          values: nodes.map((node) => node.name),
+        }
+      : null,
+  };
+
+  const perNodeColor = {
+    field: 'type',
+    type: 'nominal',
+    scale: {
+      range: colorRange,
+    },
+    legend: null,
+  };
+
+  const throughputColor = {
+    field: 'type',
+    type: 'nominal',
+    scale: {
+      range: getThroughputColorRange(nodes.length),
+    },
+    legend: null,
+  };
+
+  const perNodeTooltip = getTooltipConfig(
+    nodes.map((node) => ({
+      field: node.name,
+      type: 'quantitative',
+      title: node.name,
+      format: '.2f',
+    })),
+  );
+
+  const throughputTooltip = getTooltipConfig(
+    ((nodes) => {
+      let res = [];
+      nodes.forEach((element) => {
+        res.push({
+          field: `${element.name}-read`,
+          type: 'quantitative',
+          title: `${element.name}-read`,
+          formatType: 'throughputFormatter',
+        });
+        res.push({
+          field: `${element.name}-write`,
+          type: 'quantitative',
+          title: `${element.name}-write`,
+          format: '.2f',
+        });
+      });
+      return res;
+    })(nodes),
+  );
+
+  const lineConfig = { strokeWidth: 1.5 };
+
+  const formatNodesPromRangeForChart = useCallback(
```

Comment:
I suggest we add unit test for `formatNodesPromRangeForChart` and `formatNodesThroughputPromRangeForChart`